subprojects/core/src/main/java/org/gradle/caching/internal/tasks/BuildCacheKeyInputs.java

```diff
@@ -37,7 +36,14 @@
     private final ImmutableSortedMap<String, HashCode> inputHashes;
     private final ImmutableSortedSet<String> outputPropertyNames;
 
-    public BuildCacheKeyInputs(String taskClass, HashCode classLoaderHash, List<HashCode> actionClassLoaderHashes, ImmutableList<String> actionClassNames, ImmutableSortedMap<String, HashCode> inputHashes, ImmutableSortedSet<String> outputPropertyNames) {
+    public BuildCacheKeyInputs(
+        @Nullable String taskClass,
```

Comment:
So the problem is the construction of `NO_CACHE_KEY` here: https://github.com/gradle/gradle/blob/094a5ed82fcc9147de09afcc1970cb6260bba9f9/subprojects/core/src/main/java/org/gradle/caching/internal/tasks/DefaultTaskOutputCachingBuildCacheKeyBuilder.java#L38-L38?

`inputHashes` and `outputPropertyNames` cannot be null after all I saw.